### DISCUSSION TITLE
RE: automation on adding new repositories

### DISCUSSION THREAD

**Discussion by patrickpannn** [renovatebot/renovate#25230] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, select which platform you are using.

Bitbucket Cloud (bitbucket.org)

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

Hi,

I would like to automate the process of adding a new repository in the Config.js. Currently, our team are loading Config.js to ConfigMap and using Cronjob to run the Renovate service in Kubernetes. Every time when we want to onboard a new repository, we need to manually update the "repositories" array. I wonder if there is a way we can automate it. Any advice will be appreciated. 

### What you tried so far.

Still figuring out different ways of implementing this feature.

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by viceice** [comment]
use autodiscover?

**Comment by rarkins** [comment]
That config.JS file can read from disk or env so you could automate it that way

**Reply by patrickpannn** [reply]
is that able to load the config js into ConfigMap at runtime?

**Reply by rarkins** [reply]
It's a plain JS file
